src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/CertificateRegistry.java

```diff
@@ -5,155 +5,81 @@
         }
+        return new Certificate(certificateId, Certificate.Status.INACTIVE);
     }
 
     /**
-     * Retrieves Certificate ID from IoT Core.
-     *
-     * @param certificatePem Certificate PEM
-     * @return IoT Certificate ID or empty Optional if certificate is inactive or invalid
+     * Update certificate.
+     * @param certificate certificate object
      */
-    private Optional<String> fetchActiveCertificateId(String certificatePem) {
-        return iotAuthClient.getActiveCertificateId(certificatePem);
-    }
-
-    /**
-     * Returns IoT Certificate ID associated locally for given certificate PEM.
-     *
-     * @param certificatePem Certificate PEM
-     * @return Certificate ID or empty optional
-     */
-    private Optional<String> getAssociatedCertificateId(String certificatePem) {
-        return Optional.ofNullable(getCertificateHash(certificatePem))
-                .map(registry::get);
-    }
-
-    /**
-     * Locally caches IoT Certificate ID mapping for Certificate PEM.
-     *
-     * @param certificateId IoT Certificate ID
-     * @param certificatePem Certificate PEM
-     */
-    private void registerCertificateIdForPem(String certificateId, String certificatePem) {
-        Optional.ofNullable(getCertificateHash(certificatePem))
-                .ifPresent(certHash -> registry.put(certHash, certificateId));
-    }
-
-    /**
-     * Updates certPem <-> IoT Certificate ID association in the registry.
-     *
-     * @param certificatePem Certificate PEM
-     * @param iotCertificateId Optional of IoT Certificate ID
-     */
-    private void updateRegistryForCertificate(String certificatePem, Optional<String> iotCertificateId) {
-        if (iotCertificateId.isPresent()) {
-            registerCertificateIdForPem(iotCertificateId.get(), certificatePem);
+    public void updateCertificate(Certificate certificate) {
+        // NOTE: The storage aspect of this will change soon. But for now
+        //  only store ACTIVE certificates in the registry
+        if (certificate.getStatus() == Certificate.Status.ACTIVE) {
+            registry.put(certificate.getIotCertificateId(), certificate.getLastUpdated());
         } else {
-            clearRegistryForPem(certificatePem);
+            registry.remove(certificate.getIotCertificateId());
         }
     }
 
-    /**
-     * Removes registry entry for Certificate PEM.
-     *
-     * @param certificatePem Certificate PEM
-     */
-    private void clearRegistryForPem(String certificatePem) {
-        Optional.ofNullable(getCertificateHash(certificatePem))
-                .ifPresent(registry::remove);
-    }
-
-    /**
-     * Returns SHA-256 hash of given CertificatePem.
-     * @param certificatePem certificate pem
-     * @return Certificate hash or null if hash could not be calculated
-     */
-    private String getCertificateHash(String certificatePem) {
+    private String getCertificateId(String certificatePem) {
```

Comment:
This is ok as long as we don't expose it as public. Ideally ids should get set to `Certificate` objects when they are stored. I was thinking we would have `Certificate` objects hold a reference to their pem file and call something like

```java
Certificate cert = certificateRegistry.create(String pem); 
// create would compute the id
// Create a certificate instance with that id and store it
// return the instance
```

Reply:
The id or the method?

ID _is_ the public interface. I'm aligning with IoT core here instead of inventing out own thing